lib/command.js

```diff
@@ -1101,7 +1101,9 @@ Expecting one of '${allowedValues.join("', '")}'`);
     }
 
     // Fallback to parsing the help flag to invoke the help.
-    return this._dispatchSubcommand(subcommandName, [], [this._helpLongFlag]);
+    if (this._helpLongFlag) {
+      return this._dispatchSubcommand(subcommandName, [], [this._helpLongFlag]);
+    }
```

Comment:
Could use the flags of `subCommand` that are always inherited from the parent command but can be changed by the library user via `.helpOption()` later if they know different help flags are used in the executable.

```suggestion
    return this._dispatchSubcommand(subcommandName, [], [
      subCommand._helpLongFlag || subCommand._helpShortFlag
    ]);
```

What do you think?

Reply:
I actually don't like the `._dispatchSubcommand()` call here at all.
- It introduces an inconsistency between how commands with and without an executable handler are handled since the `preSubcommand` hooks are only called for the former.
- It makes a new `._findCommand()` call internally, which is redundant.

So I suggest
```suggestion
    this._executeSubCommand(subCommand, [
      subCommand._helpLongFlag || subCommand._helpShortFlag
    ]);
```

Reply:
No. Just fix the issue with the undefined help flag. Stay focused.

Reply:
Okay, gonna open a new PR for this.